internal/core/item_rendering.rs

```diff
@@ -95,6 +96,43 @@ pub(crate) fn is_clipping_item(item: Pin<ItemRef>) -> bool {
         || ItemRef::downcast_pin::<Clip>(item).is_some()
 }
 
+fn render_item_children(
+    renderer: &mut dyn ItemRenderer,
+    component: &ComponentRc,
+    index: isize,
+) -> VisitChildrenResult {
+    let mut actual_visitor =
+        |component: &ComponentRc, index: usize, item: Pin<ItemRef>| -> VisitChildrenResult {
+            renderer.save_state();
+
+            let (do_draw, item_geometry) = renderer.filter_item(item);
+
+            let item_origin = item_geometry.origin;
+            renderer.translate(item_origin.x, item_origin.y);
+
+            // Don't render items that are clipped, with the exception of the Clip or Flickable since
+            // they themselves clip their content.
+            if do_draw
+               || is_clipping_item(item) // HACK, the geometry of the box shadow does not include the shadow, because when the shadow is the root for repeated elements it would translate the children
+               || ItemRef::downcast_pin::<BoxShadow>(item).is_some()
+            {
+                item.as_ref().render(&mut (renderer as &mut dyn ItemRenderer));
+            }
+
+            let result = render_item_children(renderer, component, index as isize);
+
+            renderer.restore_state();
+
```

Comment:
Why so much empty lines 😃

Reply:
Pure generosity ;-)

(I like the visual separation between unrelated calls, but it's maybe extreme here - fixed now)